dataproc/src/main/java/InstantiateInlineWorkflow.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START dataproc_instantiate_inline_workflow]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.dataproc.v1.ClusterConfig;
+import com.google.cloud.dataproc.v1.GceClusterConfig;
+import com.google.cloud.dataproc.v1.HadoopJob;
+import com.google.cloud.dataproc.v1.ManagedCluster;
+import com.google.cloud.dataproc.v1.OrderedJob;
+import com.google.cloud.dataproc.v1.RegionName;
+import com.google.cloud.dataproc.v1.WorkflowMetadata;
+import com.google.cloud.dataproc.v1.WorkflowTemplate;
+import com.google.cloud.dataproc.v1.WorkflowTemplatePlacement;
+import com.google.cloud.dataproc.v1.WorkflowTemplateServiceClient;
+import com.google.cloud.dataproc.v1.WorkflowTemplateServiceSettings;
+import com.google.protobuf.Empty;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class InstantiateInlineWorkflow {
+
+  public static void InstantiateInlineWorkflow() throws IOException, InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String region = "your-project-region";
+    instantiateInlineWorkflow(projectId, region);
+  }
+
+  public static void instantiateInlineWorkflow(String projectId, String region)
+      throws IOException, InterruptedException {
+    String myEndpoint = String.format("%s-dataproc.googleapis.com:443", region);
+
+    // Configure the settings for the workflow template service client.
+    WorkflowTemplateServiceSettings workflowTemplateServiceSettings =
+        WorkflowTemplateServiceSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    // Create a workflow template service client with the configured settings. The client only
+    // needs to be created once and can be reused for multiple requests. Using a try-with-resources
+    // closes the client, but this can also be done manually with the .close() method.
+    try (WorkflowTemplateServiceClient workflowTemplateServiceClient =
+        WorkflowTemplateServiceClient.create(workflowTemplateServiceSettings)) {
+
+      // Configure the jobs within the workflow.
+      HadoopJob teragenHadoopJob =
+          HadoopJob.newBuilder()
+              .setMainJarFileUri("file:///usr/lib/hadoop-mapreduce/hadoop-mapreduce-examples.jar")
```

Comment:
Is this pre-installed?
Why is this a useful example if it's already there?

Reply:
The point we're trying to get across is how you can structure a workflow and run it against the cluster. We sometimes use these toy examples instead of say a PySpark example on GCS. 

I'm also referencing the canonical YAML example from the [docs](https://cloud.google.com/dataproc/docs/concepts/workflows/using-yamls) (which is not to say this sample couldn't be improved either).

Reply:
SG